src/components/QuestDetails/index.tsx

```diff
@@ -199,13 +201,64 @@ export default function QuestDetails({
     )
   }
 
+  const markdownComponentsProp: MarkdownProps['components'] = {
+    a: ({ href: markdownLinkHref, children, ...link }) => (
+      <a
+        target="_blank"
+        href={markdownLinkHref || ''}
+        rel="noopener noreferrer"
+        {...link}
+      >
+        <Button
+          type="link"
+          size="small"
+          spacing="xs"
+          className={styles.linkBtn}
+        >
+          {children}
+        </Button>
+      </a>
+    ),
+    ...(markdownOptions?.components || {})
+  }
+
+  const markdownAllowedElementsProp: MarkdownProps['allowedElements'] = [
+    'p',
+    'strong',
+    'b',
+    'a',
+    'i',
+    'em',
+    'ul',
+    'ol',
+    'li',
+    'blockquote',
+    'h1',
+    'h2',
+    'h3',
+    'h4',
+    'h5',
+    'h6',
+    'pre',
+    'code',
+    'hr',
+    'br',
+    ...(markdownOptions?.allowedElements || [])
+  ]
+
   let content = (
     <div className={cn(styles.rootContent, classNames?.rootContent)}>
       <div className={cn(styles.container, classNames?.content)}>
         {sticker}
         <div className={cn('title', styles.title)}>{title}</div>
         <div className={cn('body-sm', 'color-neutral-400', styles.description)}>
-          {description}
+          <Markdown
+            {...markdownOptions}
+            components={markdownComponentsProp}
+            allowedElements={markdownAllowedElementsProp}
+          >
+            {description}
+          </Markdown>
```

Comment:
how do you feel about making this a separate component? let's call it `MarkdownDescription`

We can export that component from the ui library and pass it in as a `description` prop of type element to this Quest Details component

Then we can abstract it away and we have a reusable markdown component for the future

Reply:
Sidenote but I think we could do this for the other sections of the quest details too to better decompose this component

Reply:
Yup this can be done, certainly and I think it is better as it will be standalone component, generally just like the `Address` we discussed earlier